turtlebot3_drlnav_JavAram/src/turtlebot3_drl/model/58ce_environment.txt
root@58ce3278fbf4:/home/javierac# source turtlebot3_drlnav_ws/install/setup.bash 
root@58ce3278fbf4:/home/javierac# ros2 run turtlebot3_drl environment
running on stage: 9
[ENV] Primer mapa recibido → episodio puede comenzar
R: -4.5594  MinD: 0.74   Alin: 0.09   Aturn: 0.09   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -5.2112  MinD: 0.69   Alin: -0.36  Aturn: -0.36  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.3582   MinD: 0.59   Alin: 0.21   Aturn: 0.21   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.9053  MinD: 0.41   Alin: -0.54  Aturn: -0.54  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.3557  MinD: 0.55   Alin: -0.40  Aturn: -0.40  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
[MAP] cov: 0.017 ent: 0.983 frontiers: 190
R: -0.0065  MinD: 0.52   Alin: 0.45   Aturn: 0.45   cov: 0.017 ent: 0.983 ang_grad: 0.0° frontiers: 190
R: -0.1691  MinD: 0.62   Alin: -0.37  Aturn: -0.37  cov: 0.017 ent: 0.983 ang_grad: 0.0° frontiers: 190
R: -0.7627  MinD: 0.43   Alin: -0.41  Aturn: -0.41  cov: 0.034 ent: 0.966 ang_grad: -135.9° frontiers: 354
R: -0.6067  MinD: 0.43   Alin: -0.30  Aturn: -0.30  cov: 0.034 ent: 0.966 ang_grad: 175.2° frontiers: 354
R: -0.6306  MinD: 0.50   Alin: -0.32  Aturn: -0.32  cov: 0.054 ent: 0.946 ang_grad: -177.5° frontiers: 557
R: -0.3821  MinD: 0.50   Alin: 0.06   Aturn: 0.06   cov: 0.054 ent: 0.946 ang_grad: -156.5° frontiers: 557
R: -1.2609  MinD: 0.38   Alin: -0.59  Aturn: -0.59  cov: 0.090 ent: 0.910 ang_grad: -122.0° frontiers: 862
R: -21.3997 MinD: 0.17   Alin: -0.39  Aturn: -0.39  cov: 0.090 ent: 0.910 ang_grad: -84.8° frontiers: 862
[ENV] Primer mapa recibido → episodio puede comenzar
R: -3.2489  MinD: 0.74   Alin: -0.88  Aturn: -0.88  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.3719  MinD: 0.70   Alin: -0.25  Aturn: -0.25  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -2.1760  MinD: 0.37   Alin: 0.92   Aturn: 0.92   cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 76
R: 0.4626   MinD: 0.24   Alin: 0.05   Aturn: 0.05   cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 76
R: -0.8587  MinD: 0.33   Alin: 0.67   Aturn: 0.67   cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 76
R: -0.8049  MinD: 0.29   Alin: -0.44  Aturn: -0.44  cov: 0.021 ent: 0.979 ang_grad: -118.0° frontiers: 226
R: -2.6142  MinD: 0.35   Alin: -0.84  Aturn: -0.84  cov: 0.021 ent: 0.979 ang_grad: -146.2° frontiers: 226
R: -0.4576  MinD: 0.47   Alin: 0.07   Aturn: 0.07   cov: 0.036 ent: 0.964 ang_grad: 177.6° frontiers: 386
R: -1.3124  MinD: 0.73   Alin: 0.46   Aturn: 0.46   cov: 0.036 ent: 0.964 ang_grad: 158.4° frontiers: 386
R: -1.0735  MinD: 0.66   Alin: -0.55  Aturn: -0.55  cov: 0.064 ent: 0.936 ang_grad: 109.1° frontiers: 745
R: -0.8664  MinD: 0.54   Alin: -0.56  Aturn: -0.56  cov: 0.064 ent: 0.936 ang_grad: 30.0° frontiers: 745
R: -2.6690  MinD: 0.87   Alin: -0.88  Aturn: -0.88  cov: 0.104 ent: 0.896 ang_grad: -42.1° frontiers: 1128
R: 0.0250   MinD: 0.55   Alin: -0.07  Aturn: -0.07  cov: 0.104 ent: 0.896 ang_grad: -106.7° frontiers: 1128
R: -0.8484  MinD: 0.29   Alin: 0.50   Aturn: 0.50   cov: 0.104 ent: 0.896 ang_grad: -102.3° frontiers: 1128
R: -21.2076 MinD: 0.17   Alin: 0.10   Aturn: 0.10   cov: 0.118 ent: 0.882 ang_grad: -115.4° frontiers: 1197
[ENV] Primer mapa recibido → episodio puede comenzar
R: -0.7880  MinD: 0.73   Alin: -0.35  Aturn: -0.35  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 11
R: -1.0906  MinD: 0.84   Alin: -0.44  Aturn: -0.44  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 11
R: -1.3845  MinD: 0.66   Alin: -0.65  Aturn: -0.65  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: -1.6852  MinD: 0.63   Alin: 0.84   Aturn: 0.84   cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: -0.8372  MinD: 0.84   Alin: -0.53  Aturn: -0.53  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: -0.4120  MinD: 1.09   Alin: -0.16  Aturn: -0.16  cov: 0.021 ent: 0.979 ang_grad: -158.2° frontiers: 227
R: -3.6194  MinD: 0.90   Alin: 0.92   Aturn: 0.92   cov: 0.021 ent: 0.979 ang_grad: -138.2° frontiers: 227
R: -0.9741  MinD: 0.71   Alin: -0.44  Aturn: -0.44  cov: 0.021 ent: 0.979 ang_grad: -152.8° frontiers: 227
R: -0.5496  MinD: 0.52   Alin: -0.26  Aturn: -0.26  cov: 0.021 ent: 0.979 ang_grad: -139.5° frontiers: 227
R: -3.5341  MinD: 0.46   Alin: 0.89   Aturn: 0.89   cov: 0.038 ent: 0.962 ang_grad: -151.4° frontiers: 431
R: -0.3250  MinD: 0.45   Alin: 0.13   Aturn: 0.13   cov: 0.038 ent: 0.962 ang_grad: -128.1° frontiers: 431
R: -0.3622  MinD: 0.39   Alin: 0.48   Aturn: 0.48   cov: 0.038 ent: 0.962 ang_grad: -52.4° frontiers: 431
R: -22.8143 MinD: 0.18   Alin: 0.76   Aturn: 0.76   cov: 0.056 ent: 0.944 ang_grad: -75.9° frontiers: 640
[ENV] Primer mapa recibido → episodio puede comenzar
R: -2.3269  MinD: 0.74   Alin: -0.76  Aturn: -0.76  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.0111   MinD: 0.67   Alin: -0.13  Aturn: -0.13  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -1.2135  MinD: 0.39   Alin: 0.74   Aturn: 0.74   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -3.7060  MinD: 0.26   Alin: -1.00  Aturn: -1.00  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.2818   MinD: 0.47   Alin: 0.27   Aturn: 0.27   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
[MAP] cov: 0.001 ent: 0.999 frontiers: 12
R: -0.2225  MinD: 0.60   Alin: 0.49   Aturn: 0.49   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.1368   MinD: 0.83   Alin: -0.22  Aturn: -0.22  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.3776  MinD: 0.53   Alin: -0.40  Aturn: -0.40  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.4141   MinD: 0.50   Alin: 0.14   Aturn: 0.14   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -2.2479  MinD: 0.56   Alin: -0.78  Aturn: -0.78  cov: 0.022 ent: 0.978 ang_grad: -175.5° frontiers: 208
R: -0.2877  MinD: 0.47   Alin: 0.19   Aturn: 0.19   cov: 0.022 ent: 0.978 ang_grad: -116.8° frontiers: 208
R: -0.9089  MinD: 0.44   Alin: 0.33   Aturn: 0.33   cov: 0.037 ent: 0.963 ang_grad: -160.0° frontiers: 383
R: -0.5647  MinD: 0.38   Alin: 0.18   Aturn: 0.18   cov: 0.037 ent: 0.963 ang_grad: 158.7° frontiers: 383
R: -0.3716  MinD: 0.37   Alin: -0.11  Aturn: -0.11  cov: 0.037 ent: 0.963 ang_grad: 157.1° frontiers: 383
R: -2.5271  MinD: 0.59   Alin: 0.72   Aturn: 0.72   cov: 0.057 ent: 0.943 ang_grad: 173.6° frontiers: 606
R: -2.7062  MinD: 0.79   Alin: -0.86  Aturn: -0.86  cov: 0.057 ent: 0.943 ang_grad: 123.3° frontiers: 606
R: -1.2400  MinD: 0.65   Alin: -0.55  Aturn: -0.55  cov: 0.069 ent: 0.931 ang_grad: 134.7° frontiers: 741
R: -2.1452  MinD: 0.45   Alin: 0.67   Aturn: 0.67   cov: 0.069 ent: 0.931 ang_grad: 137.2° frontiers: 741
R: -1.1512  MinD: 0.39   Alin: 0.55   Aturn: 0.55   cov: 0.084 ent: 0.916 ang_grad: 98.9° frontiers: 898
R: -1.0745  MinD: 0.68   Alin: 0.70   Aturn: 0.70   cov: 0.084 ent: 0.916 ang_grad: 35.8° frontiers: 898
R: -3.1175  MinD: 0.60   Alin: 0.92   Aturn: 0.92   cov: 0.084 ent: 0.916 ang_grad: -6.1° frontiers: 898
[ENV] Primer mapa recibido → episodio puede comenzar
R: 0.0188   MinD: 0.73   Alin: 0.11   Aturn: 0.11   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 14
R: -3.1637  MinD: 0.63   Alin: -0.87  Aturn: -0.87  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 14
R: -0.2116  MinD: 0.62   Alin: -0.35  Aturn: -0.35  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 14
R: 0.3634   MinD: 0.41   Alin: 0.21   Aturn: 0.21   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 14
R: -1.8297  MinD: 0.28   Alin: 0.86   Aturn: 0.86   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 14
R: 0.4400   MinD: 0.46   Alin: 0.14   Aturn: 0.14   cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 71
R: -0.2460  MinD: 0.72   Alin: -0.37  Aturn: -0.37  cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 71
R: -0.1604  MinD: 0.84   Alin: 0.49   Aturn: 0.49   cov: 0.017 ent: 0.983 ang_grad: 0.0° frontiers: 195
R: -0.4190  MinD: 0.68   Alin: 0.57   Aturn: 0.57   cov: 0.017 ent: 0.983 ang_grad: 0.0° frontiers: 195
R: -3.4236  MinD: 0.47   Alin: -0.97  Aturn: -0.97  cov: 0.030 ent: 0.970 ang_grad: -166.0° frontiers: 348
R: -1.1009  MinD: 0.50   Alin: -0.52  Aturn: -0.52  cov: 0.030 ent: 0.970 ang_grad: -120.7° frontiers: 348
R: -2.2057  MinD: 0.67   Alin: -0.77  Aturn: -0.77  cov: 0.030 ent: 0.970 ang_grad: -121.5° frontiers: 348
R: -1.8933  MinD: 0.88   Alin: -0.70  Aturn: -0.70  cov: 0.045 ent: 0.955 ang_grad: -153.3° frontiers: 528
R: -0.7303  MinD: 0.81   Alin: 0.25   Aturn: 0.25   cov: 0.045 ent: 0.955 ang_grad: -170.5° frontiers: 528
R: -0.3895  MinD: 0.42   Alin: -0.18  Aturn: -0.18  cov: 0.062 ent: 0.938 ang_grad: 149.9° frontiers: 740
R: -1.6386  MinD: 0.27   Alin: -0.67  Aturn: -0.67  cov: 0.062 ent: 0.938 ang_grad: -108.0° frontiers: 740
R: -1.3986  MinD: 0.35   Alin: -0.63  Aturn: -0.63  cov: 0.062 ent: 0.938 ang_grad: -60.6° frontiers: 740
R: -2.4863  MinD: 0.61   Alin: -0.83  Aturn: -0.83  cov: 0.062 ent: 0.938 ang_grad: -41.3° frontiers: 740
R: 0.2033   MinD: 0.53   Alin: 0.19   Aturn: 0.19   cov: 0.062 ent: 0.938 ang_grad: -47.5° frontiers: 740
[MAP] cov: 0.084 ent: 0.916 frontiers: 1005
R: -22.4137 MinD: 0.21   Alin: 0.85   Aturn: 0.85   cov: 0.084 ent: 0.916 ang_grad: -84.8° frontiers: 1005
R: -20.9700 MinD: 0.19   Alin: 0.16   Aturn: 0.16   cov: 0.084 ent: 0.916 ang_grad: -85.3° frontiers: 1005
[ENV] Primer mapa recibido → episodio puede comenzar
R: -2.3673  MinD: 0.73   Alin: -0.74  Aturn: -0.74  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.0198   MinD: 0.89   Alin: 0.06   Aturn: 0.06   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.5869  MinD: 0.97   Alin: 0.60   Aturn: 0.60   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.6510  MinD: 0.79   Alin: -0.48  Aturn: -0.48  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -1.5238  MinD: 0.71   Alin: -0.67  Aturn: -0.67  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.3429  MinD: 0.78   Alin: -0.05  Aturn: -0.05  cov: 0.021 ent: 0.979 ang_grad: 157.7° frontiers: 227
R: -1.5930  MinD: 0.64   Alin: -0.64  Aturn: -0.64  cov: 0.021 ent: 0.979 ang_grad: 137.7° frontiers: 227
R: -1.5094  MinD: 0.31   Alin: 0.52   Aturn: 0.52   cov: 0.021 ent: 0.979 ang_grad: 140.3° frontiers: 227
R: -22.7283 MinD: 0.19   Alin: -0.65  Aturn: -0.65  cov: 0.021 ent: 0.979 ang_grad: 141.4° frontiers: 227
[ENV] Primer mapa recibido → episodio puede comenzar
R: 0.1885   MinD: 0.75   Alin: 0.27   Aturn: 0.27   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -1.7253  MinD: 0.90   Alin: -0.69  Aturn: -0.69  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.3170  MinD: 0.96   Alin: -0.39  Aturn: -0.39  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -2.9056  MinD: 0.86   Alin: -0.89  Aturn: -0.89  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.4881  MinD: 0.55   Alin: -0.44  Aturn: -0.44  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.4580   MinD: 0.29   Alin: 0.06   Aturn: 0.06   cov: 0.006 ent: 0.994 ang_grad: 0.0° frontiers: 65
R: -23.8578 MinD: 0.19   Alin: -0.89  Aturn: -0.89  cov: 0.006 ent: 0.994 ang_grad: 0.0° frontiers: 65
[ENV] Primer mapa recibido → episodio puede comenzar
R: -0.2032  MinD: 0.74   Alin: 0.48   Aturn: 0.48   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.5654  MinD: 0.70   Alin: 0.59   Aturn: 0.59   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.2677   MinD: 0.48   Alin: -0.15  Aturn: -0.15  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -20.7817 MinD: 0.19   Alin: -0.18  Aturn: -0.18  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
[ENV] Primer mapa recibido → episodio puede comenzar
R: -3.3411  MinD: 0.75   Alin: -0.95  Aturn: -0.95  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.4061   MinD: 0.76   Alin: 0.16   Aturn: 0.16   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.4561   MinD: 0.66   Alin: 0.02   Aturn: 0.02   cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: -0.5645  MinD: 0.64   Alin: 0.60   Aturn: 0.60   cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: 0.3927   MinD: 0.40   Alin: -0.07  Aturn: -0.07  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: -23.3865 MinD: 0.18   Alin: 0.96   Aturn: 0.96   cov: 0.019 ent: 0.981 ang_grad: 0.0° frontiers: 217
[ENV] Primer mapa recibido → episodio puede comenzar
R: -2.7633  MinD: 0.74   Alin: 0.99   Aturn: 0.99   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.6344  MinD: 0.70   Alin: 0.59   Aturn: 0.59   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.7103  MinD: 0.43   Alin: -0.50  Aturn: -0.50  cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 67
R: -22.8703 MinD: 0.18   Alin: 0.87   Aturn: 0.87   cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 67
[ENV] Primer mapa recibido → episodio puede comenzar
R: -0.5891  MinD: 0.73   Alin: -0.45  Aturn: -0.45  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 11
R: -1.5231  MinD: 0.85   Alin: -0.66  Aturn: -0.66  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 11
R: -0.0397  MinD: 0.48   Alin: 0.44   Aturn: 0.44   cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: -20.7229 MinD: 0.19   Alin: 0.30   Aturn: 0.30   cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
[ENV] Primer mapa recibido → episodio puede comenzar
R: -1.5089  MinD: 0.73   Alin: -0.66  Aturn: -0.66  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 11
R: -1.3829  MinD: 0.82   Alin: 0.77   Aturn: 0.77   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 11
R: 0.1869   MinD: 0.40   Alin: -0.21  Aturn: -0.21  cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 64
R: -20.7747 MinD: 0.18   Alin: -0.19  Aturn: -0.19  cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 64
[ENV] Primer mapa recibido → episodio puede comenzar
R: -0.5200  MinD: 0.74   Alin: 0.57   Aturn: 0.57   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.1031   MinD: 0.81   Alin: 0.36   Aturn: 0.36   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.3385   MinD: 0.48   Alin: 0.25   Aturn: 0.25   cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 70
R: -23.3705 MinD: 0.19   Alin: -0.82  Aturn: -0.82  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 70
[ENV] Primer mapa recibido → episodio puede comenzar
R: -3.3252  MinD: 0.73   Alin: -0.95  Aturn: -0.95  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -2.2344  MinD: 0.83   Alin: 0.92   Aturn: 0.92   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.5331  MinD: 0.54   Alin: -0.46  Aturn: -0.46  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: -1.5171  MinD: 0.44   Alin: 0.81   Aturn: 0.81   cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: 0.0016   MinD: 0.49   Alin: -0.29  Aturn: -0.29  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 79
R: -0.4912  MinD: 0.69   Alin: 0.15   Aturn: 0.15   cov: 0.021 ent: 0.979 ang_grad: -151.4° frontiers: 233
R: -2.6415  MinD: 0.88   Alin: 0.78   Aturn: 0.78   cov: 0.021 ent: 0.979 ang_grad: -131.2° frontiers: 233
R: -1.6196  MinD: 0.75   Alin: -0.67  Aturn: -0.67  cov: 0.039 ent: 0.961 ang_grad: -100.6° frontiers: 424
R: -0.0655  MinD: 0.93   Alin: 0.42   Aturn: 0.42   cov: 0.039 ent: 0.961 ang_grad: -37.8° frontiers: 424
R: 0.3283   MinD: 1.03   Alin: -0.02  Aturn: -0.02  cov: 0.083 ent: 0.917 ang_grad: -70.8° frontiers: 835
R: -1.5687  MinD: 0.84   Alin: -0.66  Aturn: -0.66  cov: 0.083 ent: 0.917 ang_grad: -149.5° frontiers: 835
R: -0.3316  MinD: 0.65   Alin: -0.21  Aturn: -0.21  cov: 0.083 ent: 0.917 ang_grad: -172.2° frontiers: 835
R: -0.5505  MinD: 0.51   Alin: 0.25   Aturn: 0.25   cov: 0.106 ent: 0.894 ang_grad: -136.8° frontiers: 1015
R: -3.4784  MinD: 0.42   Alin: 0.88   Aturn: 0.88   cov: 0.106 ent: 0.894 ang_grad: -165.8° frontiers: 1015
R: -2.7387  MinD: 0.26   Alin: -0.86  Aturn: -0.86  cov: 0.126 ent: 0.874 ang_grad: -176.1° frontiers: 1257
R: -1.5709  MinD: 0.44   Alin: 0.54   Aturn: 0.54   cov: 0.126 ent: 0.874 ang_grad: -154.6° frontiers: 1257
R: -0.9486  MinD: 0.37   Alin: 0.36   Aturn: 0.36   cov: 0.144 ent: 0.856 ang_grad: -153.3° frontiers: 1420
R: -22.2190 MinD: 0.18   Alin: 0.45   Aturn: 0.45   cov: 0.144 ent: 0.856 ang_grad: -147.0° frontiers: 1420
[ENV] Primer mapa recibido → episodio puede comenzar
R: -0.4171  MinD: 0.73   Alin: -0.13  Aturn: -0.13  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.9928  MinD: 0.75   Alin: -0.38  Aturn: -0.38  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -1.7195  MinD: 0.55   Alin: -0.71  Aturn: -0.71  cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 64
R: -2.1905  MinD: 0.31   Alin: -0.79  Aturn: -0.79  cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 64
R: -22.4451 MinD: 0.16   Alin: 0.79   Aturn: 0.79   cov: 0.007 ent: 0.993 ang_grad: 0.0° frontiers: 64
[ENV] Primer mapa recibido → episodio puede comenzar
R: -0.5251  MinD: 0.73   Alin: -0.44  Aturn: -0.44  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -1.9489  MinD: 0.86   Alin: 0.87   Aturn: 0.87   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.4499   MinD: 0.51   Alin: 0.13   Aturn: 0.13   cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 80
R: 0.0539   MinD: 0.30   Alin: -0.27  Aturn: -0.27  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 80
R: -0.0992  MinD: 0.26   Alin: -0.33  Aturn: -0.33  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 80
R: -0.0354  MinD: 0.27   Alin: -0.29  Aturn: -0.29  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 80
R: -23.8735 MinD: 0.18   Alin: -0.89  Aturn: -0.89  cov: 0.008 ent: 0.992 ang_grad: 0.0° frontiers: 80
[ENV] Primer mapa recibido → episodio puede comenzar
R: -0.2292  MinD: 0.74   Alin: -0.35  Aturn: -0.35  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -1.0219  MinD: 0.68   Alin: 0.69   Aturn: 0.69   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -0.0820  MinD: 0.40   Alin: -0.31  Aturn: -0.31  cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: 0.4235   MinD: 0.26   Alin: 0.01   Aturn: 0.01   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12
R: -20.1244 MinD: 0.21   Alin: 0.46   Aturn: 0.46   cov: 0.001 ent: 0.999 ang_grad: 0.0° frontiers: 12

turtlebot3_drlnav_JavAram/src/turtlebot3_drl/turtlebot3_drl/common/settings.py
# ===================================================================== #
#                           GENERAL SETTINGS                            #
# ===================================================================== #

ENABLE_BACKWARD          = False    # Enable backward movement of the robot
ENABLE_STACKING          = False    # Enable processing multiple consecutive scan frames at every observation step
ENABLE_VISUAL            = False    # Meant to be used only during evaluation/testing phase
ENABLE_TRUE_RANDOM_GOALS = False    # If false, goals are selected semi-randomly from a list of known valid goal positions
ENABLE_DYNAMIC_GOALS     = False    # If true, goal difficulty (distance) is adapted according to current success rate
MODEL_STORE_INTERVAL     = 100      # Store the model weights every N episodes
GRAPH_DRAW_INTERVAL      = 10       # Draw the graph every N episodes (drawing too often will slow down training)
GRAPH_AVERAGE_REWARD     = 10       # Average the reward graph over every N episodes


# ===================================================================== #
#                         ENVIRONMENT SETTINGS                          #
# ===================================================================== #

# --- SIMULATION ENVIRONMENT SETTINGS ---
REWARD_FUNCTION = "exploration"           # Defined in reward.py
EPISODE_TIMEOUT_SECONDS = 50    # Number of seconds after which episode timeout occurs

TOPIC_SCAN = 'scan'
TOPIC_VELO = 'cmd_vel'
TOPIC_ODOM = 'odom'

EPISODE_TIMEOUT_SECONDS     = 50    # Number of seconds after which episode timeout occurs
ARENA_LENGTH                = 4.2   # meters
ARENA_WIDTH                 = 4.2   # meters
SPEED_LINEAR_MAX            = 0.3   # m/s
SPEED_ANGULAR_MAX           = 1.9   # rad/s

LIDAR_DISTANCE_CAP          = 6     # meters, for white objects can go up to 12 m
THRESHOLD_COLLISION         = 0.19  # meters around lidar reading to consider a collision
THREHSOLD_GOAL              = 0.20  # meters

IG_SUCCESS_THRESHOLD = 20.0

OBSTACLE_RADIUS             = 0.16  # meters
MAX_NUMBER_OBSTACLES        = 6
ENABLE_MOTOR_NOISE          = False # Add normally distributed noise to motor output to simulate hardware imperfections

# --- REAL ROBOT ENVIRONMENT SETTINGS ---
REAL_TOPIC_SCAN  = 'scan'
REAL_TOPIC_VELO  = 'cmd_vel'
REAL_TOPIC_ODOM  = 'odom'

# LiDAR density count your robot is providing
# NOTE: If you change this value you also have to modify
# NUM_SCAN_SAMPLES for the model in drl_environment.py
# e.g. if you increase this by 320 samples also increase
# NUM_SCAN_SAMPLES by 320 samples.
REAL_N_SCAN_SAMPLES         = 40 #500   # Number of LiDAR samples per scan. RPlidarC1 has 0.72 degree resolution -> 360/0.72 = 500

REAL_ARENA_LENGTH           = 4.2   # meters
REAL_ARENA_WIDTH            = 4.2   # meters
REAL_SPEED_LINEAR_MAX       = 0.3  # in m/s
REAL_SPEED_ANGULAR_MAX      = 1.9   # in rad/s

REAL_LIDAR_CORRECTION       = 0.003 # meters, subtracted from the real LiDAR values. Not sure about this value. I think is substracted to avoid overestimation of distance to obstacle. Set as the accuracy.
REAL_LIDAR_DISTANCE_CAP     = 6     # meters, scan distances are capped this value
REAL_THRESHOLD_COLLISION    = 0.19  # meters, robot size ~0.34, minimum distance to an object that counts as a collision
REAL_THRESHOLD_GOAL         = 0.35  # meters, minimum distance to goal that counts as reaching the goal


# ===================================================================== #
#                       DRL ALGORITHM SETTINGS                          #
# ===================================================================== #

# DRL parameters
REWARD_FUNCTION = "exploration"       # Defined in reward.py
ACTION_SIZE     = 2         # Not used for DQN, see DQN_ACTION_SIZE
HIDDEN_SIZE     = 512       # Number of neurons in hidden layers

BATCH_SIZE      = 128       # Number of samples per training batch
BUFFER_SIZE     = 1000000   # Number of samples stored in replay buffer before FIFO
DISCOUNT_FACTOR = 0.99
LEARNING_RATE   = 0.003
TAU             = 0.003

OBSERVE_STEPS   = 15000     # At training start random actions are taken for N steps for better exploration
STEP_TIME       = 0.1 #0.01      # Delay between steps, can be set to 0
EPSILON_DECAY   = 0.9995    # Epsilon decay per step
EPSILON_MINIMUM = 0.05

# DQN parameters
DQN_ACTION_SIZE = 5
TARGET_UPDATE_FREQUENCY = 1000

# DDPG parameters

# TD3 parameters
POLICY_NOISE            = 0.2
POLICY_NOISE_CLIP       = 0.5
POLICY_UPDATE_FREQUENCY = 2

# Stacking
STACK_DEPTH = 3             # Number of subsequent frames processed per step
FRAME_SKIP  = 4             # Number of frames skipped in between subsequent frames

# Episode outcome enumeration
UNKNOWN = 0
SUCCESS = 1
COLLISION_WALL = 2
COLLISION_OBSTACLE = 3
TIMEOUT = 4
TUMBLE = 5
RESULTS_NUM = 6